controllers/users/updateAvatar.js

```diff
@@ -0,0 +1,32 @@
+import fs from 'fs/promises';
+
+import User from '../../models/user.js';
+import { cropAndResize } from '../../helpers/index.js';
+
+export const updateAvatar = async (req, res, next) => {
+    try {
+        const { id } = req.user
+
+        if (!req.file) {
+            return res.status(400).json({ message: 'No file uploaded' })
+        }
+
+        const { filename, path: tempPath } = req.file
+        await cropAndResize(tempPath)
+        const newPath = `public/avatars/${filename}`
+        await fs.rename(tempPath, newPath)
+
+        const avatarURL = `/avatars/${filename}`
+
+        const user = await User.findById(id)
+        await User.findByIdAndUpdate(user._id, { avatarURL })
```

Comment:
Ta ścieżka powinna być autoryzowana, zobacz że user już jest w req.user -> linia 8. 

dlatego to co jest w 21 lini nie jest potrzebne. Wykorzystaj tutaj to co masz